contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -239,11 +245,57 @@ def _transmit_without_retry(self, envelopes):
                 text,
             )
 
+    def enable_standard_metrics(self):
+        # Sets up stats related objects to begin
+        # recording standard metrics
+        stats_ = stats.stats
+        view_manager = stats_.view_manager
+        stats_recorder = stats_.stats_recorder
+
+        # Standard metrics uses a separate instance of MeasurementMap
+        # from regular metrics but still shares the same underlying
+        # map data in the context. This way, the processing during
+        # record is handled through a separate data structure but
+        # the storage only uses one single data structure
+        # Uniqueness should be based off the name of the view and
+        # measure. A warning message in the logs will be shown
+        # if there are duplicate names
+        self.standard_metrics_map = stats_recorder.new_measurement_map()
+
+        free_memory_measure = measure_module.MeasureInt("Free memory",
+            "Amount of free memory in bytes",
+            "bytes")
+        self.standard_metrics_measure_map[StandardMetricType.FREE_MEMORY] = free_memory_measure
+        free_memory_view = view_module.View(StandardMetricType.FREE_MEMORY,
+            "Amount of free memory in bytes",
+            [],
+            free_memory_measure,
+            aggregation_module.LastValueAggregation())
+        view_manager.register_view(free_memory_view)
+
+    def record_standard_metrics(self):
+        # Function called periodically to record standard metrics
+        vmem = psutil.virtual_memory()
+        free_memory = vmem.free
+        free_measure = self.standard_metrics_measure_map[StandardMetricType.FREE_MEMORY]
+        if free_measure is not None:
+            self.standard_metrics_map.measure_int_put(free_measure, free_memory)
+        self.standard_metrics_map.record()
+
 
 def new_metrics_exporter(**options):
-    options = Options(**options)
-    exporter = MetricsExporter(options=options)
+    options_ = Options(**options)
+    exporter = MetricsExporter(options=options_)
+    if options_.enable_standard_metrics:
+        exporter.enable_standard_metrics()
+        # Separate thread for recording of metrics
+        transport.get_recorder_thread(exporter.record_standard_metrics,
+                                      interval=options_.export_interval)
```

Comment:
Not sure if the recording interval should be the same as export interval. What are the other SDKs doing today?

Reply:
In the Node SDK, the intervals for recording and exporting are different and can be configured. I was using the same interval because I was thinking of simplicity for the user so they do not have to configure anything in options. Perhaps derive the recording interval from the export interval (maybe interval-%age), as long as the recording happens prior.

Reply:
@reyang We can also add another option in Options. What do you think?

Reply:
I wonder how we plan to do this in OpenCensus/OpenTelemetry, it is not supposed to be exporter specific I guess?

Reply:
If it won't be exporter specific, we can abstract out Options to have OpenCensus default behaviour and then implement exporter specific behaviour. Just an idea.